Source/JavaScriptCore/bytecode/InlineCacheCompiler.h

```diff
@@ -194,14 +193,14 @@ class InlineCacheHandler final : public RefCounted<InlineCacheHandler> {
 
 private:
     InlineCacheHandler() = default;
-    InlineCacheHandler(Ref<PolymorphicAccessJITStubRoutine>&&, std::unique_ptr<WatchpointsOnStructureStubInfo>&&);
+    InlineCacheHandler(Ref<PolymorphicAccessJITStubRoutine>&&, std::unique_ptr<StructureStubInfoClearingWatchpoint>&&);
 
     static Ref<InlineCacheHandler> createSlowPath(VM&, AccessType);
 
     CodePtr<JITStubRoutinePtrTag> m_callTarget;
     CodePtr<JITStubRoutinePtrTag> m_jumpTarget;
     RefPtr<PolymorphicAccessJITStubRoutine> m_stubRoutine;
-    std::unique_ptr<WatchpointsOnStructureStubInfo> m_watchpoints;
+    std::unique_ptr<StructureStubInfoClearingWatchpoint> m_watchpoint;
```

Comment:
Why not have `InlineCacheHandler` subclass `StructureStubInfoClearingWatchpoint` or embed the watchpoint inline (not sure if that's possible)?

Reply:
Right now, we are just using the current existing mechanism. It is nice if we can refactor after seeing all things get working well. But for now, I think it is too early to do some refactoring.